app-frontend/src/app/components/importModal/importModal.controller.js

```diff
@@ -0,0 +1,271 @@
+    }
+
+    setCurrentStep(step) {
+        this.currentStep = step;
+        this.onStepEnter();
+    }
+
+    gotoPreviousStep() {
+        if (this.hasPreviousStep()) {
+            this.setCurrentStep(this.steps[this.getCurrentStepIndex() - 1]);
+        }
+    }
+
+    gotoNextStep() {
+        if (this.hasNextStep()) {
+            this.setCurrentStep(this.steps[this.getCurrentStepIndex() + 1]);
+        }
+    }
+
+    gotoStep(step) {
+        const stepIndex = this.steps.indexOf(step);
+        if (stepIndex) {
+            this.setCurrentStep(this.steps[stepIndex]);
+        }
+    }
+
+    gotoSceneBrowser() {
+        if (this.project) {
+            this.close();
+            this.$state.go('browse', {projectid: this.project.id});
+        }
+    }
+
+    createProject() {
+        return this.projectService.createProject(this.projectBuffer.name);
+    }
+
+    validateProjectName() {
+        if (this.projectBuffer.name) {
+            this.showProjectCreateError = false;
+            return true;
+        }
+        this.projectCreateErrorText = 'A name is needed for your new project';
+        this.showProjectCreateError = true;
+        return false;
+    }
+
+    verifyFileCount() {
+        if (this.selectedFiles.length) {
+            this.allowNext = true;
+        } else {
+            this.allowNext = false;
+        }
+    }
+
+    handleNext() {
+        if (this.allowNext) {
+            if (this.currentStepIs('LOCAL_UPLOAD')) {
+                this.gotoNextStep();
+                this.allowNext = false;
+                this.startUpload();
+            } else {
+                this.gotoNextStep();
+            }
+        }
+    }
+
+    onStepEnter() {
+        if (this.currentStepIs('LOCAL_UPLOAD')) {
+            this.verifyFileCount();
+        } else if (this.currentStepIs('IMPORT')) {
+            this.allowNext = true;
+        }
+    }
+
+    startUpload() {
+        this.userService
+            .getCurrentUser()
+            .then(this.createUpload.bind(this))
+            .then(upload => {
+                this.upload = upload;
+                return upload;
+            })
+            .then(this.getUploadCredentials.bind(this))
+            .then(this.sendFiles.bind(this));
+    }
+
+
+    createUpload(user) {
+        return this.uploadService.create({
+            files: this.selectedFiles.map(f => f.name),
+            datasource: this.resolve.datasource.id,
+            fileType: 'GEOTIFF',
+            uploadType: 'LOCAL',
+            uploadStatus: 'UPLOADING',
+            visibility: 'PRIVATE',
+            organizationId: user.organizationId,
+            metadata: {}
+        });
+    }
+
+    getUploadCredentials(upload) {
+        return this.uploadService.credentials(upload);
+    }
+
+    sendFiles(credentialData) {
+        this.uploadedFileCount = 0;
+        const parser = document.createElement('a');
+        parser.href = credentialData.bucketPath;
```

Comment:
Was going to suggest [this](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL), but it's not supported in IE so this seems like the way to go.